### PR TITLE
Allow for specifying custom meta-data for the cloud-init image.

### PR DESCRIPTION
**Pull request by night199uk**
Works exactly the same way as specifying custom user-data does.

### CODE REVIEW

File: libvirt/resource_cloud_init.go [hunk 1]
```diff
@@ -34,6 +34,11 @@ func resourceCloudInit() *schema.Resource {
 				Optional: true,
 				ForceNew: true,
 			},
+			"meta_data": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```
Comment (monstermunchkin) [inline libvirt/resource_cloud_init.go]: You should also add `Computed: true` here because `meta_data` is is populated without explicitly specifying it.

**Comment by night199uk** [comment]
Rebased, and updated as per comments. Sorry for the delays, hopefully this is GTG.

**Comment by MalloZup** [comment]
thx for contrib @night199uk !! 
i would like to have a testacc for this feature. 
 Otherwise looks good to me

**Comment by MalloZup** [comment]
https://github.com/dmacvicar/terraform-provider-libvirt/issues/384

File: libvirt/cloudinit_def.go [hunk 2]
```diff
@@ -197,11 +198,16 @@ func (ci *defCloudInit) createFiles() (string, error) {
...
 	}
 
-	metadata, err := yaml.Marshal(&ci.MetaData)
+	mergedMetaData, err := mergeMetaDataIntoMetaDataRaw(ci.MetaData, ci.MetaDataRaw)
 	if err != nil {
-		return "", fmt.Errorf("Error dumping cloudinit's meta data: %s", err)
+		return "", fmt.Errorf("Error merging Metadata with MetadataRaw: %v", err)
 	}
-	if err = ioutil.WriteFile(filepath.Join(tmpDir, metaData), metadata, os.ModePerm); err != nil {
+	metadata := fmt.Sprintf("%s", mergedMetaData)
```
Comment (flavio) [inline libvirt/cloudinit_def.go]: Maybe I didn't have enough coffee yet this morning, but do we actually need to do that? `mergedMetaData` is already a `string`.

**Comment by MalloZup** [comment]
I close this pr in favor of https://github.com/dmacvicar/terraform-provider-libvirt/pull/410 which will make the metadata field more flexible and stable. It will adress also this pr. Thanks a lot to everyone, have a nice day and journey

**Pull request closed without merging**